crates/red_knot_python_semantic/src/types.rs

```diff
@@ -804,6 +828,24 @@ impl<'db> Type<'db> {
                 .iter()
                 .any(|&elem_ty| elem_ty.is_assignable_to(db, ty)),
 
+            // A typevar is always assignable to itself, and is never assignable to any other
+            // typevar (since there is no guarantee that they will be specialized to the same
+            // type).
+            // TODO: Two distinct typevars that are both bounded by the same final class are
+            // subtypes of each other.
```

Comment:
@sharkdp I think your comment about `Never` applies here, too:

```py
@final
class F: ...

def f[T: F, U: F](t: list[T], u: list[U]) -> None: ...
```

since `f[F, Never]` and `f[Never, F]` are valid specializations.